components/GitHubEventAction.tsx

```diff
@@ -14,8 +30,8 @@ function GitHubEventAction({type, payload}: GitHubEventActionProps): JSX.Element
   switch(type){
     case 'CreateEvent':
     case 'DeleteEvent': {
-      const target = payload?.ref;
-      const targetType = payload?.ref_type;
+      const target = (payload as DeleteEvent).ref;
```

Comment:
Maybe consider casting before this to prevent duplication + ease of reading? (And other items)